src/datomic_toolbox/core.clj

```diff
@@ -21,97 +20,28 @@
 
 (defn db [] (d/db (connection)))
 
+(defn tempid
+  ([]  (d/tempid (partition)))
+  ([n] (d/tempid (partition) n)))
+
 (defn transact [tx-data]
```

Comment:
Yeah, more or less. I'm saying that I'd like to isolate (as much as possible) the stateful configuration stuff to `core`. So anything that relies on that stateful config should be in `core` and call stateless fns in other namespaces as needed. (I'm open to revisiting this architecture if others don't like it, though. Now's a good time to change it.)

So moving the single-arity version of `swap-tx!` into `core` seems OK to me.

We should also think about what to do with `transaction/deftx-data-fn` since it calls `(core/tempid)` (relying on the stateful `partition` setting). Would it make sense to put the current arity into `core` and have it call a version in `transaction` that takes the partition as its first argument? Or should we stuff the partition into the `params` arg? Thoughts, @tie-rack?